programs/zstdcli.c

```diff
@@ -1213,6 +1227,25 @@ int main(int const argCount, const char* argv[])
         if (adaptMin > cLevel) cLevel = adaptMin;
         if (adaptMax < cLevel) cLevel = adaptMax;
 
+        if (showDefaultCParams) {
+            size_t fileNb;
+            static const char* ZSTD_strategyMap[9 + 1] = { "", "ZSTD_fast",
+                "ZSTD_dfast", "ZSTD_greedy", "ZSTD_lazy", "ZSTD_lazy2", "ZSTD_btlazy2",
+                "ZSTD_btopt", "ZSTD_btultra", "ZSTD_btultra2"};
+            for (fileNb = 0; fileNb < (size_t)filenames->tableSize; fileNb++) {
+                const size_t fileSize = (size_t)UTIL_getFileSize(filenames->fileNames[fileNb]);
```

Comment:
We should check for failures here. It is possible that the first filename is a regular file, but the second isn't.

Reply:
When we don't know the file size, instead of just erroring, we can say `size: unknown` and give the "generic" parameters.

Reply:
Ah okay. good point